ecl/hql/hqlgram.y

```diff
@@ -6725,6 +6727,36 @@ primexpr1
                             Owned<ITypeInfo> ltype = lexpr->getType();
                             Owned<IHqlExpression> locale = (ltype->getTypeCode() == type_varstring) ? lexpr.getLink() : createValue(no_implicitcast, makeVarStringType(ltype->getStringLen()), lexpr.getLink());
                             $$.setExpr(createValue(no_unicodeorder, makeIntType(4, true), $3.getExpr(), $5.getExpr(), locale.getLink(), $9.getExpr()));
+                         }
+    | LIKELY '(' booleanExpr ')'
+                        {
+                            $$.inherit($3);
+                            $$.setPosition($1);
+                        }
+    | LIKELY '(' booleanExpr ',' expression ')'
+                        {
+                            parser->normalizeExpression($5, type_real, true);
+                            OwnedHqlExpr probability = $5.getExpr();
+                            double p = probability->queryValue()->getRealValue();
+                            if (p <= 0.0 || p >= 1.0)
```

Comment:
Unfortunately you can't check this here - it will crash if the value is a parameter (which is passed a constant value).  Much simpler to not check at the moment!